salt/modules/etcd_mod.py

```diff
@@ -79,10 +79,7 @@ def get_(key, recurse=False, profile=None, **kwargs):
         salt myminion etcd.get /path/to/key host=127.0.0.1 port=2379
     """
     client = __utils__["etcd_util.get_conn"](__opts__, profile, **kwargs)
-    if recurse:
-        return client.tree(key)
-    else:
-        return client.get(key, recurse=recurse)
+    return client.get(key, recurse=recurse)
```

Comment:
**question** I've just been punching some etcd code - do you know what the difference between get and read is? :thinking: (get didn't work for me, but read did)

Reply:
So, `get` just calls `read`, but calling read allows you to do watching of values.  I just added code to fix `get` to work with `recurse=True` (basically deferring that responsibility to `EtcdClient.tree`).  If you pull the newest changes and it still doesn't work, let me know.

Reply:
Well, my issue was specifically in the cache code... but this looks like it's actually something that we have in our util (which I'm :suspect: about - that it's not just one thing - or two different things, but it's "three" different things)

Reply:
Yup, the `get` code in the util has been refactored.  The choice was to go with either `read` with `recursive=True` or `tree`, and I chose `tree` because, in my opinion, it is more intuitive and easy to work with.